### DISCUSSION TITLE
Why does  `DeepGlobeLandCoverDataModule` not make use of its defined augmentations?

### DISCUSSION THREAD

**Discussion by matlorr** [torchgeo/torchgeo#3073] [Q&A]
Hello,

I was trying to train a `trainer.SemanticSegmentationTask` using the `DeepGlobeLandCoverDataModule` and I noticed that my sanity checks and validation checks were throwing dimensionality errors when trying to calculate cross entropy. After unsuccessfully defining and applying custom transformations onto the underlying datasets, I discovered that `DeepGlobeLandCoverDataModule` possesses its own transformations/augmentation, which are not used when setting up the datasets. Please see [here](https://torchgeo.readthedocs.io/en/stable/_modules/torchgeo/datamodules/deepglobelandcover.html) for more details. Applying the internally defined transformations/augmentation did solve my issue.

Can anyone tell me please why the transformations are hidden and not used? I had to create a custom data loader class inheriting from `DeepGlobeLandCoverDataModule`  so I could modify the `setup` method to use the already defined transformations. Did I miss something and did I perhaps apply the `DeepGlobeLandCoverDataModule`  incorrectly?

**Comment by adamjstewart** [comment]
Your dimensionality bug is very likely related to #2920 and may be fixed by #3028. There was a bug in `_ExtractPatches` preventing it from working correctly. Maybe @blaz-r @hkristen can check.

The augmentations are used, but are applied on each mini-batch during `on_after_batch_transfer`, not directly passed to the dataset as a `transform`.

**Reply by blaz-r** [reply]
This could be linked to that transforms issue, could you maybe share the exact exception @matlorr ?

**Comment by adamjstewart** [comment]
I'm actually unable to reproduce any issues on main. All images and masks have the same dimensions. No time series is involved, so `_ExtractPatches` shouldn't be the problem. Can you provide code to reproduce the issue you're experiencing?
